src/engine/controls/cuecontrol.cpp

```diff
@@ -1845,9 +1858,15 @@ void HotcueControl::slotHotcuePositionChanged(double newPosition) {
     emit hotcuePositionChanged(this, newPosition);
 }
 
-void HotcueControl::slotHotcueColorChanged(double newColorId) {
-    m_pCue->setColor(Color::kPredefinedColorsSet.predefinedColorFromId(newColorId));
-    emit hotcueColorChanged(this, newColorId);
+void HotcueControl::slotHotcueColorChanged(double newColor) {
+    mixxx::RgbColor::optional_t color = doubleToRgbColor(newColor);
+    if (!color) {
+        qWarning() << "slotHotcueColorChanged got invalid value:" << newColor;
+        return;
+    }
+
+    m_pCue->setColor(*color);
```

Comment:
if (!m_pCue) {
    return; 
}
To avoid segfaults changing unassigned cues.

Reply:
Good catch!